src/node_zlib.cc

```diff
@@ -553,6 +556,9 @@ class ZCtx : public AsyncWrap {
 
     if (ctx->err_ != Z_OK) {
       ZCtx::Error(ctx, "Init error");
+      if (dictionary != nullptr)
+        delete[] dictionary;
+      return false;
```

Comment:
I’d have to admit I prefer the solution suggested by @cjihrig in https://github.com/nodejs/node/issues/13082#issuecomment-302246222 (or just move this error-checking block to the end of the `Init()` function, after `ctx->init_done_ = true`)